### PR TITLE
[RetroPlayer] Flush/sync audio

### PR DESCRIPTION
**Pull request by GTechAlpha**
## Description
Flush audio (re-sync) when delay is too high. Requires 8097ba8 or timing issues will occur.

## Motivation and Context
Over-buffering of audio causing significant delay.

## How Has This Been Tested?
Device: Odroid-C2
Emulator: PCSX-REARMED

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by GTechAlpha** [comment]
@garbear

**Review by garbear — APPROVED** [review]

File: xbmc/cores/RetroPlayer/streams/RetroPlayerAudio.cpp [hunk 1]
```diff
@@ -118,8 +122,19 @@ void CRetroPlayerAudio::AddStreamData(const StreamPacket &packet)
...
     if (m_pAudioStream)
     {
+      const double delay = m_pAudioStream->GetDelay();
+
       const size_t frameSize = m_pAudioStream->GetChannelCount() * (CAEUtil::DataFormatToBits(m_pAudioStream->GetDataFormat()) >> 3);
-      m_pAudioStream->AddData(&audioPacket.data, 0, static_cast<unsigned int>(audioPacket.size / frameSize));
+
+      const unsigned int frameCount = static_cast<unsigned int>(audioPacket.size / frameSize);
+
+      if (delay > 0.3)
```
Comment (garbear) [inline xbmc/cores/RetroPlayer/streams/RetroPlayerAudio.cpp]: I missed this when reviewing, can you move the magic number to the top of the file right after `using namespace RETRO;`? Avoid using a #define

File: xbmc/cores/RetroPlayer/streams/RetroPlayerAudio.cpp [hunk 2]
```diff
@@ -118,8 +122,19 @@ void CRetroPlayerAudio::AddStreamData(const StreamPacket &packet)
   {
     if (m_pAudioStream)
     {
+      const double delay = m_pAudioStream->GetDelay();
```
Comment (garbear) [inline xbmc/cores/RetroPlayer/streams/RetroPlayerAudio.cpp]: Always include the units in the var name or its documentation, e.g. `delaySecs`

**Comment by garbear** [comment]
Test builds are available in the test builds thread: https://forum.kodi.tv/showthread.php?tid=173361

**Comment by GTechAlpha** [comment]
You're killing me  :)  At some point, you have to read the code anyway.

**Comment by garbear** [comment]
sorry to grill :) i believe in absolute clarity for readers and maintainers, because at the end of the day that's who it's about. code doesn't change the world, people do.

**Comment by GTechAlpha** [comment]
:cry: That's deep.
>code doesn't change the world, people do.

For now :wink:    https://github.com/garbear/xbmc/issues/89

**Comment by garbear** [comment]
jenkins build this please

**Pull request merged**